# Exploring Altmetrics and Times Cited in Publications - EDA using SweetViz and Analysis UsingSciPy
Python Script prompts for file selection, brings it in as dataframe df, utilizes SweetViz to conduct exploratory data analysis

This specifically process data exported from Dimensions database

In [1]:
#prompts to select data file
import pandas as pd
import tkinter as tk
from tkinter import filedialog


def create_dataframe_from_file():
    # Create a Tkinter window
    root = tk.Tk()
    root.withdraw()  # Hide the main window

    # Prompt the user to select a file
    file_path = filedialog.askopenfilename(title="Select a file")

    if file_path:
        # Read the file into a DataFrame
        df = pd.read_csv(file_path)  # Adjust this line for different file types (e.g., pd.read_excel for Excel files)

        return df
    else:
        print("No file selected.")
        return None

if __name__ == "__main__":
    # Create a DataFrame df from the selected file
    df = create_dataframe_from_file()

    if df is not None:
        print("DataFrame df created successfully:")
        print(df.head())
    else:
        print("DataFrame df creation aborted.")



/Users/sarahngo/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3338: DtypeWarning: Columns (1,9,19,20,21,22,23,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232,233,234,235,236,237,238,239,240,241,242,243,244,245,246,247,248,249,250,251,252,253,254,255,256,257,258,

DataFrame df created successfully:
   Publication ID         PMID        PMCID  \
0  pub.1169565255  3.84594e+07          NaN   
1  pub.1169527971  3.84565e+07          NaN   
2  pub.1169539976   3.8452e+07          NaN   
3  pub.1169505694  3.84512e+07          NaN   
4  pub.1168682532  3.83297e+07  PMC10853864   

                                               Title  \
0  How Primary Care Clinicians Process Patient De...   
1  The U.S. deprescribing research network: Catal...   
2  Development of a liver disease-Specific large ...   
3  Primary Palliative Care in Urology: Quality Im...   
4  Trends in Phenylephrine and Pseudoephedrine Sa...   

                                             Funding  \
0  Educational discretionary funds, Clinician Edu...   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4  Funding/Support: This study was funded through...   

        

In [2]:
df.head()


Publication ID         PMID        PMCID  \
0  pub.1169565255  3.84594e+07          NaN   
1  pub.1169527971  3.84565e+07          NaN   
2  pub.1169539976   3.8452e+07          NaN   
3  pub.1169505694  3.84512e+07          NaN   
4  pub.1168682532  3.83297e+07  PMC10853864   

                                               Title  \
0  How Primary Care Clinicians Process Patient De...   
1  The U.S. deprescribing research network: Catal...   
2  Development of a liver disease-Specific large ...   
3  Primary Palliative Care in Urology: Quality Im...   
4  Trends in Phenylephrine and Pseudoephedrine Sa...   

                                             Funding  \
0  Educational discretionary funds, Clinician Edu...   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4  Funding/Support: This study was funded through...   

                                 Source title  \
0        Journal of General Internal Medicine   
1  Journal of the American Geriatrics Society   
2                                  Hepatology   
3                            Urology Practice   
4                                        JAMA   

                            Publisher  \
0                     Springer Nature   
1                               Wiley   
2                      Wolters Kluwer   
3                      Wolters Kluwer   
4  American Medical Association (AMA)   

                                          MeSH terms Publication date PubYear  \
0                                                NaN           3/8/24    2024   
1                                                NaN           3/8/24    2024   
2                                                NaN           3/7/24    2024   
3                                                NaN           3/7/24    2024   
4  Commerce; Phenylephrine; Pseudoephedrine; Unit...           3/5/24    2024   

   ... Unnamed: 2144 Unnamed: 2145 Unnamed: 2146 Unnamed: 2147 Unnamed: 2148  \
0  ...           NaN           NaN           NaN           NaN           NaN   
1  ...           NaN           NaN           NaN           NaN           NaN   
2  ...           NaN           NaN           NaN           NaN           NaN   
3  ...           NaN           NaN           NaN           NaN           NaN   
4  ...           NaN           NaN           NaN           NaN           NaN   

  Unnamed: 2149 Unnamed: 2150 Unnamed: 2151 Unnamed: 2152 Unnamed: 2153  
0           NaN           NaN           NaN           NaN           NaN  
1           NaN           NaN           NaN           NaN           NaN  
2           NaN           NaN           NaN           NaN           NaN  
3           NaN           NaN           NaN           NaN           NaN  
4           NaN           NaN           NaN           NaN           NaN  

[5 rows x 2154 columns]

In [3]:
#check what columns were imported, then drop them if cells do not hold data
for idx, column_name in enumerate(df.columns):
    print(f"Column {idx}: {column_name}")

Column 0: Publication ID
Column 1: PMID
Column 2: PMCID
Column 3: Title
Column 4: Funding
Column 5: Source title
Column 6: Publisher
Column 7: MeSH terms
Column 8: Publication date
Column 9: PubYear
Column 10: Publication date (online)
Column 11: Publication date (print)
Column 12: Open Access
Column 13: Document Type
Column 14: Corresponding Authors
Column 15: Research Organizations - standardized
Column 16: Funder
Column 17: Funder Group
Column 18: Supporting Grants
Column 19: Times cited
Column 20: Recent citations
Column 21: RCR
Column 22: FCR
Column 23: Altmetric
Column 24: RCDC Categories
Column 25: Unnamed: 25
Column 26: Unnamed: 26
Column 27: Unnamed: 27
Column 28: Unnamed: 28
Column 29: Unnamed: 29
Column 30: Unnamed: 30
Column 31: Unnamed: 31
Column 32: Unnamed: 32
Column 33: Unnamed: 33
Column 34: Unnamed: 34
Column 35: Unnamed: 35
Column 36: Unnamed: 36
Column 37: Unnamed: 37
Column 38: Unnamed: 38
Column 39: Unnamed: 39
Column 40: Unnamed: 40
Column 41: Unnamed: 41
Column 

Column 1471: Unnamed: 1471
Column 1472: Unnamed: 1472
Column 1473: Unnamed: 1473
Column 1474: Unnamed: 1474
Column 1475: Unnamed: 1475
Column 1476: Unnamed: 1476
Column 1477: Unnamed: 1477
Column 1478: Unnamed: 1478
Column 1479: Unnamed: 1479
Column 1480: Unnamed: 1480
Column 1481: Unnamed: 1481
Column 1482: Unnamed: 1482
Column 1483: Unnamed: 1483
Column 1484: Unnamed: 1484
Column 1485: Unnamed: 1485
Column 1486: Unnamed: 1486
Column 1487: Unnamed: 1487
Column 1488: Unnamed: 1488
Column 1489: Unnamed: 1489
Column 1490: Unnamed: 1490
Column 1491: Unnamed: 1491
Column 1492: Unnamed: 1492
Column 1493: Unnamed: 1493
Column 1494: Unnamed: 1494
Column 1495: Unnamed: 1495
Column 1496: Unnamed: 1496
Column 1497: Unnamed: 1497
Column 1498: Unnamed: 1498
Column 1499: Unnamed: 1499
Column 1500: Unnamed: 1500
Column 1501: Unnamed: 1501
Column 1502: Unnamed: 1502
Column 1503: Unnamed: 1503
Column 1504: Unnamed: 1504
Column 1505: Unnamed: 1505
Column 1506: Unnamed: 1506
Column 1507: Unnamed: 1507
C

In [4]:
df.drop(columns=df.columns[25:2153], inplace=True)

In [5]:
df['PMID'] = df['PMID'].astype(str)
df['Title'] = df['Title'].astype(str)
df['Publication date'] = pd.to_numeric(df['Publication date'], errors='coerce')
df['PubYear'] = pd.to_numeric(df['PubYear'], errors='coerce')
df['Publication date (print)'] = pd.to_numeric(df['Publication date (print)'], errors='coerce')
df['Publication date (online)'] = pd.to_numeric(df['Publication date (online)'], errors='coerce')
df['Supporting Grants'] = df['Supporting Grants'].astype(str)
df['Times cited'] = pd.to_numeric(df['Times cited'], errors='coerce')
df['Recent citations'] = pd.to_numeric(df['Recent citations'], errors='coerce')
df['RCR'] = pd.to_numeric(df['RCR'], errors='coerce')
df['FCR'] = pd.to_numeric(df['FCR'], errors='coerce')
df['Altmetric'] = pd.to_numeric(df['Altmetric'], errors='coerce')

In [6]:
#installing the sweetviz library 
!pip install sweetviz

In [7]:
# import the required libraries 
import sweetviz as sv
from sklearn.datasets import fetch_california_housing
from sklearn.model_selection import train_test_split

print("SweetViz Version : {}".format(sv.__version__))


SweetViz Version : 2.3.1


In [8]:
# analyzing the dataset
report = sv.analyze(df,
                    pairwise_analysis= "on")


In [9]:
#show the report
report.show_html('Dimensions.html')

Report Dimensions.html was generated! NOTEBOOK/COLAB USERS: the web browser MAY not pop up, regardless, the report IS saved in your notebook/colab files.


In [10]:
# finding correlation using spearman rank stat test
!pip install scipy


In [17]:
# Convert NaN to 0
df.fillna(0, inplace=True)

print(df)

      Publication ID        PMID        PMCID  \
0     pub.1169565255  38459411.0            0   
1     pub.1169527971  38456495.0            0   
2     pub.1169539976  38451962.0            0   
3     pub.1169505694  38451199.0            0   
4     pub.1168682532  38329748.0  PMC10853864   
...              ...         ...          ...   
5003  pub.1041520622  29067294.0   PMC5644276   
5004  pub.1030602508  26492216.0   PMC4869989   
5005  pub.1025517269  27019745.0   PMC4789688   
5006  pub.1105347393  29975036.0            0   
5007  pub.1079273291  27198102.0            0   

                                                  Title  \
0     How Primary Care Clinicians Process Patient De...   
1     The U.S. deprescribing research network: Catal...   
2     Development of a liver disease-Specific large ...   
3     Primary Palliative Care in Urology: Quality Im...   
4     Trends in Phenylephrine and Pseudoephedrine Sa...   
...                                                 ...  

In [18]:
import scipy.stats

# Calculate Spearman rank correlation coefficient
rho, p_value = scipy.stats.spearmanr(df['Altmetric'], df['Times cited'])

print("Spearman rank correlation coefficient:", rho)
print("p-value:", p_value)

Spearman rank correlation coefficient: 0.6168680685558957
p-value: 0.0


# Spearman rank correlation Results:
Spearman rank correlation coefficient: 0.6168680685558957
p-value: 0.0

# Interpretation
A Spearman rank correlation coefficient of 0.616 indicates a moderately strong positive monotonic relationship between the two variables. Here's what each part of the output means:

Spearman rank correlation coefficient: This value measures the strength and direction of the monotonic relationship between the two variables. In this case, the coefficient is 0.616, indicating a positive correlation.

p-value: This is the probability of observing the data given that the null hypothesis is true, which in this case is that there is no correlation between the two variables. A p-value of 0.0 indicates that the observed correlation is statistically significant, meaning it's highly unlikely to occur by random chance. Typically, if the p-value is below a certain threshold (often 0.05), you reject the null hypothesis and conclude that there is a significant correlation between the variables.
    

In [19]:
# Calculate Kendall's tau coefficient
tau, p_value = scipy.stats.kendalltau(df['Altmetric'], df['Times cited'])

print("Kendall's tau coefficient:", tau)
print("p-value:", p_value)

Kendall's tau coefficient: 0.47529194524392815
p-value: 0.0


# Kendall's tau coefficient Results:

Kendall's tau coefficient: 0.47529194524392815
p-value: 0.0

# Interpretation
A Kendall's tau coefficient of 0.475 and a p-value of 0.0 suggest a moderate positive association between the altmetrics and times cited, which is statistically significant.

+1 indicates a perfect agreement,
-1 indicates a perfect disagreement,
0 indicates no association.

In [20]:
from scipy.stats import linregress
import numpy as np

# Perform linear regression
slope, intercept, r_value, p_value, std_err = linregress(df['Altmetric'], df['Times cited'])

# Print results
print("Slope:", slope)
print("Intercept:", intercept)
print("R-squared:", r_value**2)
print("p-value:", p_value)
print("Standard error:", std_err)


Slope: 0.23648598009991972
Intercept: 9.494442174369635
R-squared: 0.18907313095714318
p-value: 3.9324038615598984e-230
Standard error: 0.006922068833234106


# Linear Regression Results:
Slope: 0.23648598009991972
Intercept: 9.494442174369635
R-squared: 0.18907313095714318
p-value: 3.9324038615598984e-230
Standard error: 0.006922068833234106

# Interpretation
Null Hypothesis: there is no linear relationship between independent variables Times Cited and dependent variable Altmetrics

Slope: the slope is approximately 0.2365, which means that for each unit increase in Times Cited, Altmetrics is expected to increase by about 0.2365 units, all else being equal.

Intercept: The intercept of the regression line is the value of dependent variable Altmetrics when Times Cited is zero. Here, the intercept is approximately 9.4944, suggesting that when Times Cited is zero, the expected value of Altmetrics is around 9.4944.

R-squared: is approximately 0.1891, indicating that about 18.91% of the variance in Altmetrics can be explained by the linear relationship with Times Cited. While this value is relatively low, it still suggests some degree of association between the variables.

p-value: The very small p-value (approximately 3.932×10−230) indicates strong evidence against the null hypothesis, suggesting that there is indeed a significant linear relationship between the variables.

Standard error: It is approximately 0.0069, indicating the typical amount that the observed values deviate from the predicted values by the regression model.